python/lsst/ip/isr/crosstalk.py

```diff
@@ -675,18 +699,44 @@ def subtractCrosstalk(self, thisExposure, sourceExposure=None, crosstalkCoeffs=N
             coeffsSqr = coeffsSqr.transpose()
             mi2 = mi.clone()
             mi2.scaledMultiplies(1.0, mi)
+
         for ss, sAmp in enumerate(sourceDetector):
-            sImage = subtrahend[sAmp.getBBox() if isTrimmed else sAmp.getRawDataBBox()]
+            if fullAmplifier:
+                sImage = subtrahend[sAmp.getBBox() if isTrimmed else sAmp.getRawBBox()]
+            elif parallelOverscan:
+                if detectorConfig is not None:
+                    ampConfig = detectorConfig.getOverscanAmpConfig(sAmp)
+                    if not ampConfig.doParallelOverscanCrosstalk:
+                        # Skip crosstalk correction for this amplifier.
+                        continue
+
+                sImage = subtrahend[sAmp.getRawParallelOverscanBBox()]
+            else:
+                sImage = subtrahend[sAmp.getBBox() if isTrimmed else sAmp.getRawDataBBox()]
             for tt, tAmp in enumerate(detector):
                 if coeffs[ss, tt] == 0.0:
                     continue
-                tImage = self.extractAmp(mi, tAmp, sAmp, isTrimmed)
+                tImage = self.extractAmp(
+                    mi,
+                    tAmp,
+                    sAmp,
+                    isTrimmed=isTrimmed,
+                    fullAmplifier=fullAmplifier,
+                    parallelOverscan=parallelOverscan,
+                )
                 tImage.getMask().getArray()[:] &= crosstalk  # Remove all other masks
                 tImage -= backgrounds[tt]
                 sImage.scaledPlus(coeffs[ss, tt], tImage)
                 # Add the nonlinear term
                 if doSqrCrosstalk:
-                    tImageSqr = self.extractAmp(mi2, tAmp, sAmp, isTrimmed)
+                    tImageSqr = self.extractAmp(
```

Comment:
Can you add a comment here pointing out that `mi2` has already been squared, so this is correct as written?